renderlib/io/FileReaderTIFF.cpp

```diff
@@ -281,6 +296,103 @@ readTiffDimensions(TIFF* tiff, const std::string filepath, VolumeDimensions& dim
+    if (scifiometadata.empty()) {
+      LOG_ERROR << "Unexpected bad or empty TIFF SCIFIO metadata";
+      return false;
+    }
+
+    std::vector<std::string> axes;
+    auto iter = scifiometadata.find("axes");
+    if (iter != scifiometadata.end()) {
+      split((*iter).second, ',', axes);
+    } else {
+      LOG_WARNING << "Failed to read axes of SCIFIO TIFF: '" << filepath << "'";
+    }
+    std::vector<std::string> scales;
+    iter = scifiometadata.find("scales");
+    if (iter != scifiometadata.end()) {
+      split((*iter).second, ',', scales);
+    } else {
+      LOG_WARNING << "Failed to read scales of SCIFIO TIFF: '" << filepath << "'";
+    }
+    std::vector<std::string> lengths;
+    iter = scifiometadata.find("lengths");
+    if (iter != scifiometadata.end()) {
+      split((*iter).second, ',', lengths);
+    } else {
+      LOG_WARNING << "Failed to read lengths of SCIFIO TIFF: '" << filepath << "'";
+    }
+    std::vector<std::string> units;
+    iter = scifiometadata.find("units");
+    if (iter != scifiometadata.end()) {
+      split((*iter).second, ',', units);
+    } else {
+      LOG_WARNING << "Failed to read units of SCIFIO TIFF: '" << filepath << "'";
+    }
+
+    if (lengths.size() != units.size() || scales.size() != axes.size() || scales.size() != units.size()) {
+      LOG_ERROR << "SCIFIO TIFF metadata has inconsistent counts of lengths,units,axes,scales";
+    }
+
+    for (size_t i = 0; i < axes.size(); ++i) {
+      std::string axis = axes[i];
+      double scale = std::stod(scales[i]);
+      std::string unit = units[i];
+      int length = std::stoi(lengths[i]);
+      if (axis == "X") {
+        // check consistency.
+        if (sizeX != length) {
+          LOG_WARNING << "Inconsistent X size " << length << " in SCIFIO TIFF: '" << filepath << "'";
+        }
+        physicalSizeX = scale;
+      } else if (axis == "Y") {
+        if (sizeY != length) {
+          LOG_WARNING << "Inconsistent Y size " << length << " in SCIFIO TIFF: '" << filepath << "'";
+        }
+        physicalSizeY = scale;
+      } else if (axis == "Z") {
+        sizeZ = length;
+        physicalSizeZ = scale;
+      } else if (axis == "Channel") {
+        sizeC = length;
+      } else if (axis == "Time") {
+        sizeT = length;
+      } else if (axis == "Unknown" && axes.size() == 3) {
+        // make a guess and use the last axis as Z
+        sizeZ = length;
+        physicalSizeZ = scale;
+      }
+    }
+    iter = scifiometadata.find("channels");
+    if (iter != scifiometadata.end()) {
+      int nch = std::stoi((*iter).second);
+      if (sizeC != nch) {
+        LOG_WARNING << "Inconsistent number of channels " << nch << " in SCIFIO TIFF: '" << filepath << "'";
+      }
+    } else {
+      LOG_WARNING << "Failed to read number of channels of SCIFIO TIFF: '" << filepath << "'";
+    }
+
+    for (uint32_t i = 0; i < sizeC; ++i) {
+      channelNames.push_back(std::to_string(i));
+    }
```

Comment:
should you fail out here on fatal errors?  Or just expect to crash soon, having outputted diagnostic messages?

Reply:
For now it's enough to have logged the error.  Most of the errors have some kind of graceful recovery that doesn't crash (for example: don't set the value we are trying to read, only load one channel instead of N channels, or return an empty image).    The docs for Agave say if something went wrong and you don't know why, look for the logfile first.